src/server.c

```diff
@@ -1314,19 +1298,20 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
         atomicGet(server.stat_net_repl_input_bytes, stat_net_repl_input_bytes);
         atomicGet(server.stat_net_repl_output_bytes, stat_net_repl_output_bytes);
         monotime current_time = getMonotonicUs();
-        trackInstantaneousRateMetric(STATS_METRIC_COMMAND, server.stat_numcommands, current_time);
-        trackInstantaneousRateMetric(STATS_METRIC_NET_INPUT, stat_net_input_bytes + stat_net_repl_input_bytes,
-                                     current_time);
-        trackInstantaneousRateMetric(STATS_METRIC_NET_OUTPUT,
-                                     stat_net_output_bytes + stat_net_repl_output_bytes, current_time);
-        trackInstantaneousRateMetric(STATS_METRIC_NET_INPUT_REPLICATION, stat_net_repl_input_bytes,
-                                     current_time);
-        trackInstantaneousRateMetric(STATS_METRIC_NET_OUTPUT_REPLICATION, stat_net_repl_output_bytes,
-                                     current_time);
-        trackInstantaneousRateMetric(STATS_METRIC_EL_CYCLE, server.duration_stats[EL_DURATION_TYPE_EL].cnt,
-                                     current_time);
-        trackInstantaneousAvgMetric(STATS_METRIC_EL_DURATION, server.duration_stats[EL_DURATION_TYPE_EL].cnt,
-                                    server.duration_stats[EL_DURATION_TYPE_EL].sum);
+        long long factor = 1000000;  // us
+        trackInstantaneousMetric(STATS_METRIC_COMMAND, server.stat_numcommands, current_time, factor);
+        trackInstantaneousMetric(STATS_METRIC_NET_INPUT, stat_net_input_bytes + stat_net_repl_input_bytes,
+                                 current_time, factor);
+        trackInstantaneousMetric(STATS_METRIC_NET_OUTPUT, stat_net_output_bytes + stat_net_repl_output_bytes,
+                                 current_time, factor);
+        trackInstantaneousMetric(STATS_METRIC_NET_INPUT_REPLICATION, stat_net_repl_input_bytes, current_time,
+                                 factor);
+        trackInstantaneousMetric(STATS_METRIC_NET_OUTPUT_REPLICATION, stat_net_repl_output_bytes,
+                                 current_time, factor);
+        trackInstantaneousMetric(STATS_METRIC_EL_CYCLE, server.duration_stats[EL_DURATION_TYPE_EL].cnt,
+                                 current_time, factor);
+        trackInstantaneousMetric(STATS_METRIC_EL_DURATION, server.duration_stats[EL_DURATION_TYPE_EL].sum,
+                                 server.duration_stats[EL_DURATION_TYPE_EL].cnt, factor);
```

Comment:
shouldn't this be `1` instead of `factor`?

i.e. in the other metric we count rate per second, so the factor scales up the delta time to a full second.
but here we measure an average duration per cycle so `sum` is the duration and `cnt` is the cycle count.
`sum` is in microseconds, and we want the result to remain in microseconds, so `factor` should be 1, right?

Reply:
……I was being careless